### PR TITLE
Change HTTPHeaderDict vals intertal representation to list

### PR DESCRIPTION
**Pull request by yetingsky**
Just using list will make the code clear and not break the API

### CODE REVIEW

File: urllib3/_collections.py [hunk 1]
```diff
@@ -278,10 +268,7 @@ def __repr__(self):
     def _copy_from(self, other):
         for key in other:
             val = other.getlist(key)
-            if isinstance(val, list):
-                # Don't need to convert tuples
-                val = list(val)
-            self._container[key.lower()] = [key] + val
+            self._container[key.lower()] = [key] + list(val)
```
Comment (sigmavirus24) [inline urllib3/_collections.py]: So I would rather be defensive here. The other object *might not* be a HTTPHeaderDict and could potentially not return a list. If it returns, for example, a string then we'll be doing:

```
>>> list('str')
['s', 't', 'r']
```

Comment (yetingsky) [inline urllib3/_collections.py]: I think here `list(val)` will just be fine. [getlist](https://github.com/yetingsky/urllib3/blob/master/urllib3/_collections.py#L257) will always return list, according to docstring and implementation.
Actually, `isinstance(val, list)` may never reach

**Comment by yetingsky** [comment]
when a head only has one value, using tuple. but when a head has many value(like cookie header), using list because tuple can not append... This does not make sense to me... Maybe be consistent will be better.

File: urllib3/_collections.py (hunk 1 above)
Comment (sigmavirus24) [inline urllib3/_collections.py]: @yetingsky a `HTTPHeaderDict` is not the only thing we might be copying from though. I don't trust other implementations to be 100% correct.

**Comment by yetingsky** [comment]
@sigmavirus24 :+1: you are right, l have leaved that part unchanged

**Pull request merged**